lib/view_component/compiler.rb

```diff
@@ -24,34 +24,40 @@ def compile(raise_errors: false, force: false)
       return if @component == ViewComponent::Base
 
       @lock.synchronize do
-        # this check is duplicated so that concurrent compile calls can still
-        # early exit
+        # This check is duplicated so that concurrent compile calls can still
+        # early exit before we instrument.
         return if compiled? && !force
 
-        gather_templates
+        ActiveSupport::Notifications.instrument(
```

Comment:
After thinking about how we might write the documentation for this feature, my gut tells me that we might be better served with something simpler.

What if we instead added a no-op public `after_compile` method to `Base` and call it inside `https://github.com/ViewComponent/view_component/blob/main/lib/view_component/base.rb#L539`? That way, consumers could define their own `after_compile` that takes additional actions.

Reply:
The `after_compile` hook seems much cleaner and avoids the Rails 7.1 issues. I’ll update the PR to use that approach.

Reply:
Perhaps using ActiveSupport::Callbacks would make it easier to inject multiple extensions into the base gem